pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -790,6 +790,25 @@ func ProviderFromSource(packageSource string) (plugin.Provider, error) {
 				return nil, fmt.Errorf("could not find installed plugin %s, did you mean ./%[1]s: %w", descriptor.Name, err)
 			}
 
+			if descriptor.SubDir() != "" {
+				path, err := descriptor.DirPath()
+				if err != nil {
+					return nil, err
+				}
+				info, statErr := os.Stat(filepath.Join(path, descriptor.SubDir()))
+				if statErr == nil && info.IsDir() {
```

Comment:
An alternative would be to always run dep installation on linking. For node it will always be very fast due to package-lock.json, for python it's reasonably fast too, since pip will see that the packages are already present in the venv. Or we could put a marker `.deps-installed` marker file inside the directory after we've run installation, and look for that to skip installation.

I think I'm ok with the current approach too though.

Reply:
Yeah, I went for this because it's how we already do the plugin installation, so I think it makes sense to keep the dependency installation here as well.